client/extensions/woocommerce/app/promotions/fields/promotion-applies-to-field/index.js

```diff
@@ -102,24 +104,66 @@ class PromotionAppliesToField extends React.Component {
 		this.initializeValue( appliesToType );
 	};
 
+	onProductIdChange = ( productId ) => {
+		const { edit } = this.props;
+		edit( 'appliesTo', { productIds: [ productId ] } );
+	};
+
+	onProductIdsChange = ( productIds ) => {
+		const { edit } = this.props;
+		edit( 'appliesTo', { productIds } );
+	};
+
+	renderProducts = ( appliesTo, singular ) => {
+		const productIds = ( appliesTo && appliesTo.productIds ? appliesTo.productIds : [] );
+
+		return (
+			<ProductSearch
+				value={ productIds }
+				onChange={ ( singular ? this.onProductIdChange : this.onProductIdsChange ) }
+				singular={ singular }
+			/>
+		);
+	};
+
+	renderProductCategories = ( appliesTo, singular, edit ) => {
+		return (
+			<AppliesToFilteredList
+				appliesToType={ 'productCategoryIds' }
+				singular={ singular }
```

Comment:
Can you have a single product category sale? Maybe it's worth simplifying `AppliesToFilteredList` to just a `ProductCategorySearch` component?

Reply:
I was thinking the same thing. I think it would make sense to create that component and move the code to it.